src/components/hook/useTimeTravel.js

```diff
@@ -0,0 +1,33 @@
+import { useState, useEffect } from "react";
+import React from "react";
+
+export default function useTimeTravel() {
+  const [date, setDate] = useState([]);
+  const [index, setIndex] = useState(null);
+  const [current, setCurrent] = useState("");
+
+  useEffect(() => {
+    if (index !== null) {
+      setCurrent(date[index]);
+    }
+  }, [index]);
+
+  const save = (event) => {
+    setCurrent(event.target.value);
+    setDate((prevState) => [...prevState, event.target.value]);
+    setIndex(date.length);
```

Comment:
this should be `setIndex(prevIndex => prevIndex + 1)` using date.length will always send the user to the end of the array